File: sdk/nodejs/examples/sandbox-session.ts
import { AgentKernel } from "agentkernel";

const client = new AgentKernel();

// Create a sandbox session — auto-removed when scope exits
await using sb = await client.sandbox("demo", { image: "python:3.12-alpine" });

// Install a package
await sb.run(["pip", "install", "numpy"]);

// Run code
const result = await sb.run([
  "python3",
  "-c",
  "import numpy; print(f'numpy {numpy.__version__}')",
]);
console.log(result.output);

// Sandbox is automatically removed here

File: sdk/python/examples/sandbox_session.py
"""Sandbox session example for agentkernel Python SDK."""

from agentkernel import AgentKernel

client = AgentKernel()

# Create a sandbox session — auto-removed when context manager exits
with client.sandbox("demo", image="python:3.12-alpine") as sb:
    # Install a package
    sb.run(["pip", "install", "numpy"])

    # Run code
    result = sb.run(["python3", "-c", "import numpy; print(f'numpy {numpy.__version__}')"])
    print(result.output)

# Sandbox is automatically removed here
